modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,1106 @@
+
+
+        Quat<_Tp> operator-() const;
+
+        bool operator==(const Quat<_Tp>&) const;
+
+        Quat<_Tp> operator+(const Quat<_Tp>&) const;
+
+        Quat<_Tp>& operator+=(const Quat<_Tp>&);
+
+        Quat<_Tp> operator-(const Quat<_Tp>&) const;
+
+        Quat<_Tp>& operator-=(const Quat<_Tp>&);
+
+        Quat<_Tp>& operator*=(const Quat<_Tp>&);
+
+        Quat<_Tp>& operator*=(const _Tp&);
+
+        Quat<_Tp> operator*(const Quat<_Tp>&) const;
+
+        Quat<_Tp> operator/(const _Tp&) const;
+
+        Quat<_Tp> operator/(const Quat<_Tp>&) const;
+
+        Quat<_Tp>& operator/=(const _Tp&);
+
+        Quat<_Tp>& operator/=(const Quat<_Tp>&);
+
+        _Tp& operator[](std::size_t n);
+
+        const _Tp& operator[](std::size_t n) const;
+
+        template <typename S>
+        friend Quat<S> cv::operator*(const S, const Quat<S>&);
+
+        template <typename S>
+        friend Quat<S> cv::operator*(const Quat<S>&, const S);
+
+        template <typename S>
+        friend std::ostream& cv::operator<<(std::ostream&, const Quat<S>&);
+
+        _Tp w, x, y, z;
+    };
+    template <typename T>
+    Quat<T> rvec2Quat(Vec<T, 3> &rvec);
+
+    template <typename T>
+    Quat<T> inv(const Quat<T> &q, bool assumeUnit=false);
+
+    template <typename T>
+    Quat<T> sinh(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> cosh(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> tanh(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> sin(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> cos(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> tan(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> asinh(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> acosh(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> atanh(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> asin(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> acos(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> atan(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> power(const Quat<T> &q, const Quat<T> &p, bool assumeUnit=false);
+
+    template <typename T>
+    Quat<T> exp(const Quat<T> &q);
+
+    template <typename T>
+    Quat<T> log(const Quat<T> &q, bool assumeUnit=false);
+
+    template <typename T, typename _T>
+    Quat<T> power(const Quat<T>& q, _T x, bool assumeUnit=false);
+
+    template <typename T>
+    Quat<T> crossProduct(const Quat<T> &p, const Quat<T> &q);
+
+    template <typename S>
+    Quat<S> sqrt(const Quat<S> &q, bool assumeUnit=false);
+
+    template <typename S>
+    Quat<S> operator*(const S, const Quat<S>&);
+
+    template <typename S>
+    Quat<S> operator*(const Quat<S>&, const S);
+
+    template <typename S>
+    std::ostream& operator<<(std::ostream&, const Quat<S>&);
+
+    using Quatd = Quat<double>;
+    using Quatf = Quat<float>;
+    using std::sin;
```

Comment:
Is it ok to have these `using std::*` in this header for the user?

Reply:
No, it is not ok. It should be fixed.

Reply:
We overload the sin function for queternion in Quat class, and we must use std::sin(angle) in the cpp file, otherwise, an error will be reported
```
error: no matching function for call to ‘cv::Quat<double>::sin(double)’
```
In the quaternion file, we could call sin, cos in the way of std::sin and std::cos. but some cpp files which are already mergerd in opencv source use sin without std::sin to call sin(double).

To have `using std::*` is the only way I know to solve this problem. 

Or is there a better solution to solve this problem?

Reply:
Use `std::sin` (or `::std::sin` in worst case)

Reply:
You mean to keep this way of my usage without changing the code, right?

Reply:
I mean remove "using std::*" from headers.
Use full namespace names where it is required.

Reply:
OK. 
I tested it just now and confused with no error mentioned above during compile...